yams-py/yams-py/Patcher.py

```diff
@@ -0,0 +1,79 @@
+import json
+import os
+import platform
+import shutil
+import subprocess
+import sys
+import tempfile
+from pathlib import Path
+
+# TODO: revise this.
+yams_path = os.fspath(Path(__file__).with_name(name="yams"))
+sys.path.append(yams_path)
+from pythonnet import load
+load("coreclr")
+import clr
+clr.AddReference("YAMS-LIB")
+from YAMS_LIB import Patcher
+
+
+def patch_game(input_path, output_path, patch_data):
+    # Copy to input dir to temp dir first to do operations there
+    progress_update("Copying to temporary path...", -1)
+    tempdir = tempfile.TemporaryDirectory()
+    shutil.copytree(input_path, tempdir.name, dirs_exist_ok=True)
+
+    # Get data.win path. Both of these *need* to be strings, as otherwise patcher won't accept them.
+    output_data_win_path: str = os.fspath(self._get_data_win_path(tempdir.name))
+    input_data_win_path: str = shutil.move(output_data_win_path, output_data_win_path + "_orig")
+
+    # Temp write patch_data into json file for yams later
+    progress_update("Creating json file...", -1)
+    json_file = tempfile.NamedTemporaryFile(mode='w+', delete=False)
+    json_file.write(json.dumps(patch_data))
+    json_file.close()
+
+    # AM2RLauncher installations usually have a profile.xml file. For less confusion, remove it if it exists
+    if Path.exists(Path(tempdir.name).joinpath("profile.xml")):
```

Comment:
`exists`/`unlink` are methods of `Path`. You should just make an object and call methods :)